### PR TITLE
chore: house authority PDA and whitelisted house keys

### PR DESCRIPTION
**Pull request by kdevdivvy**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/divvy-bet exports TVL: 
        
 
 
    solana                    495.97 k
    
    total                    495.97 k

**Comment by llamatester** [comment]
Error while running adapter at projects/divvy-bet: 
        
 
  
    
    
    TypeError: Max seed length exceeded
    Truncated error stack:
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@solana/web3.js/lib/index.cjs.js:269:15
        at Array.forEach (<anonymous>)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:32:31)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:143:9
        at async Promise.all (index 0)
        at async Promise.all (index 1)

**Comment by llamatester** [comment]
Error while running adapter at projects/divvy-bet: 
        
 
  
    
    
    TypeError: Max seed length exceeded
    Truncated error stack:
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@solana/web3.js/lib/index.cjs.js:269:15
        at Array.forEach (<anonymous>)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:32:31)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:143:9
        at async Promise.all (index 0)
        at async Promise.all (index 1)

**Comment by llamatester** [comment]
The adapter at projects/divvy-bet exports TVL: 
        
 
 
    solana                    686.98 k
    
    total                    686.98 k

File: projects/divvy-bet/index.js [hunk 1]
```diff
@@ -1,24 +1,49 @@
...
+        [Buffer.from("House Authority"), house.publicKey.toBuffer()],
+        programId
+      )[0];
+      return [house.account.currency, owner];
+    });
+  const solOwners = [
+    PublicKey.findProgramAddressSync(
+      [
+        Buffer.from("House Authority"),
+        new PublicKey("haus4QCRft9QKj2iQL9MUiQ2P6uYnWVyhFEZ3QsSa5U").toBuffer(),
```
Comment (kdevdivvy) [inline projects/divvy-bet/index.js:36]: @g1nt0ki FYI - this should be tracked as `solOwner` because it stores native SOL, not SPL wrapped SOL. Thanks!

**Comment by llamatester** [comment]
Error while running adapter at projects/divvy-bet: 
        
 
  
    
    
    TypeError: Cannot read properties of undefined (reading 'toBuffer')
    Truncated error stack:
        at PublicKey.createProgramAddressSync (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@solana/web3.js/lib/index.cjs.js:273:58)
        at findHouseAuthorityAddress (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:7:22)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:32:21
        at Array.map (<anonymous>)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:31:6)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:143:9
        at async Promise.all (index 0)

**Comment by llamatester** [comment]
Error while running adapter at projects/divvy-bet: 
        
 
  
    
    
    TypeError: Cannot read properties of undefined (reading 'toBuffer')
    Truncated error stack:
        at PublicKey.createProgramAddressSync (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@solana/web3.js/lib/index.cjs.js:273:58)
        at findHouseAuthorityAddress (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:7:22)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:32:21
        at Array.map (<anonymous>)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:31:6)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:143:9
        at async Promise.all (index 0)

**Comment by llamatester** [comment]
Error while running adapter at projects/divvy-bet: 
        
 
  
    
    
    ReferenceError: DIVVY_PROGRAM_ID is not defined
    Truncated error stack:
    
        at findHouseAuthorityAddress (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:9:7)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:31:21
        at Array.map (<anonymous>)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/divvy-bet/index.js:30:6)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:143:9
        at async Promise.all (index 0)
        at async Promise.all (index 1)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:186:3

File: projects/divvy-bet/index.js [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
+      },
+    },
+  ]);
+  const tokensAndOwners = houses
+    .filter((house) => house.account.currency.toBase58() !== NATIVE_MINT)
+    .map((house) => {
+      const owner = findHouseAuthorityAddress(house.publicKey, programId);
+      return [house.account.currency, owner];
+    });
+  const solOwners = houses
```
Comment (kdevdivvy) [inline projects/divvy-bet/index.js:34]: @g1nt0ki FYI - this should be tracked as solOwner because it stores native SOL, not SPL wrapped SOL. Thanks!

**Comment by llamatester** [comment]
The adapter at projects/divvy-bet exports TVL: 
        
 
 
    solana                    613.43 k
    
    total                    613.43 k

**Comment by g1nt0ki** [comment]
looks good, thanks

**Pull request merged**